api/api.py

```diff
@@ -20,6 +20,35 @@ class LiveUpdateStep:
   """
   pass
 
+class PortForward:
+  """
+  Specifications for running and displaying a Kubernetes port-forward.
+
+  By default, the host for a port-forward is ``localhost``. This can be changed with
+  the ``--host`` flag when invoking Tilt via the CLI.
+
+  Attributes:
+    local_port (int): the local port to forward traffic to.
+    container_port (int, optional): if provided, the container port to foward traffic _from_. If not provided, Tilt will forward traffic from ``local_port``, if exposed, and otherwise, from the first default container port. E.g.: ``PortForward(1111)`` forwards traffic from container port 1111 (if exposed; otherwise first default container port) to ``localhost:1111``.
+    link_text (str, optional): if provided, the text of this URL when displayed in the Web UI. This parameter can be useful for disambiguating between multiple port-forwards on a single resource, e.g. labeling one link "App" and one "Debugger." If not given, the Web UI displays the URL itself (e.g. "localhost:8888").
+  """
+  def __init__(self, local_port: int,
```

Comment:
the other classes in our api tend to have a function (e.g., `blob()`, `run()`, `sync()`) rather than a class constructor (which has slightly different semantics). is there a reason why you chose a different approach here?

Reply:
oh! none in particular, I can use a constructor function! (what are the semantic differences of note, out of curiosity?)

Reply:
the capital vs lowercase is the most salient (but important!) one. 

i believe that in core python, there are some differences in what operations they support

starlark has no built-in classes, so you'd have to simulate it with structs